jdaviz/core/tools.py

```diff
@@ -338,6 +338,30 @@ def is_visible(self):
         return len([m for m in self.viewer.figure.marks if isinstance(m, SpectralLine)]) > 0
 
 
+@viewer_tool
+class StretchBounds(CheckableTool):
+    icon = os.path.join(ICON_DIR, 'line_select.svg')
+    tool_id = 'jdaviz:stretch_bounds'
+    action_text = 'Set Stretch VMin and VMax'
+    tool_tip = 'Set Stretch VMin (left-click) and VMax (alt+left-click)'
+
+    def __init__(self, viewer, **kwargs):
+        super().__init__(viewer, **kwargs)
+
+    def activate(self):
+        self.viewer.add_event_callback(self.on_mouse_event,
+                                       events=['click'])
+
+    def deactivate(self):
+        self.viewer.remove_event_callback(self.on_mouse_event)
+
+    def on_mouse_event(self, data):
+        if data["altKey"]:
+            self.viewer._plugin.stretch_vmax_value = data['domain']['x']
+        else:
+            self.viewer._plugin.stretch_vmin_value = data['domain']['x']
```

Comment:
I still think we might want a single tool to modify spline knots as well, in which case moving the closest on click might make more sense.  But this probably works until we make that decision.  That said, alt doesn't exist on macs... can we use the same right-click/ctrl-click that is used for the toolbar?

Reply:
It's option-click on Mac, I forgot it's labeled differently since my Mac keyboard has both words on the key. The viewer area doesn't capture the mouse context in the same way as the toolbar does, so right clicking there gives you the Jupyter right-click menu.

Reply:
I think we can prevent the context/jupyter menu from the vue-side.  I can take a quick stab at it if you want

Reply:
and after looking a little more, it seems that the right-click isn't even being processed by bqplot and the event on the vue-side does not contain data coordinates, so emitting a custom event won't do what we need anyways.  We'd need support for this upstream if we want right-click, but I think sticking to alt for now is probably fine and we can revisit when we add support for editable splines (which might either want right-click of their own or to be merged with this and just act on the nearest point 🤷‍♂️ )